project-koku/korekuta-operator-go
# COST-546: Get initial upload in place

## TODO: 
* Add cert verification
* Add tests
## Testing Instructions: 
1. Either run the operator locally or in the cluster from this branch 
2. Deploy the CR with the default token authentication: 
```
make deploy-cr
```
3. Watch the operator logs - reach out to me for a tar.gz (or you can use any tar.gz to upload).
* If you are running locally - create the `/tmp/cost-mgmt-operator-reports/` directory copy the tarfile to that location. 
* If you are running in the cluster you can copy the tarfile to the pod using this command (You will replace the podname with your specific podname): 
```
kubectl cp payload1.tar.gz korekuta-operator-go-controller-manager-5ffd598cbd-678fp:/tmp/cost-mgmt-operator-reports/ -c manager
```
4. Watch the operator logs 
5. Repeat with basic authentication: 
```
make deploy-cr USER=$YOUR_USERNAME PASS=$YOUR_PASSWORD AUTH=basic
```
6. For fun - use an incorrect password to see the error logs for 401 and also check the status of the CR to make sure that the status has all of the correct information. 

Reach out with any questions
